applications/serial_lte_modem/src/slm_at_udp_proxy.c

```diff
@@ -191,24 +201,61 @@ static int do_udp_client_connect(const char *url, uint16_t port)
 			ret = -errno;
 			goto cli_exit;
 		}
+
+		if (using_cid) {
+			if (setsockopt(proxy.sock, SOL_TLS, TLS_DTLS_CID,
+							&proxy.dtls_cid, sizeof(proxy.dtls_cid))) {
+				ret = -errno;
+				LOG_WRN("Setting DTLS CID (%d) failed: %d", proxy.dtls_cid, ret);
+				goto cli_exit;
+			}
+		}
 	}
 
 	/* Connect to remote host */
-	ret = util_resolve_host(0, url, port, proxy.family, &sa);
+	ret = util_resolve_host(0, url, port, proxy.family, __log_current_const_data, &sa);
 	if (ret) {
-		LOG_ERR("getaddrinfo() error: %s", gai_strerror(ret));
 		goto cli_exit;
 	}
-	if (sa.sa_family == AF_INET) {
-		ret = connect(proxy.sock, &sa, sizeof(struct sockaddr_in));
-	} else {
-		ret = connect(proxy.sock, &sa, sizeof(struct sockaddr_in6));
-	}
+
+	LOG_INF("Connecting to %s:%d (%s)...", url, port, using_dtls ? "DTLS" : "UDP");
+	const size_t size = (sa.sa_family == AF_INET) ?
+		sizeof(struct sockaddr_in) : sizeof(struct sockaddr_in6);
+
+	ret = connect(proxy.sock, &sa, size);
 	if (ret < 0) {
 		LOG_ERR("connect() failed: %d", -errno);
 		ret = -errno;
 		goto cli_exit;
 	}
+	LOG_INF("Connected.");
+
+	if (using_cid) {
+		int cid_status;
+		socklen_t cid_status_size = sizeof(cid_status);
+
+		/* Check the connection identifier status, and fail
+		 * if it does not correspond to what was requested.
+		 */
+		ret = getsockopt(proxy.sock, SOL_TLS, TLS_DTLS_CID_STATUS,
+							&cid_status, &cid_status_size);
+		if (ret) {
+			LOG_ERR("Getting DTLS CID status failed: %d", ret);
+			ret = -errno;
+			goto cli_exit;
+		} else {
+			if (proxy.dtls_cid == TLS_DTLS_CID_ENABLED
+				&& (cid_status == TLS_DTLS_CID_STATUS_DISABLED
+					|| cid_status == TLS_DTLS_CID_STATUS_DOWNLINK)) {
+				LOG_ERR("DTLS CID status (%d) not satisfactory for"
+					" requested DTLS CID (%d).", cid_status, proxy.dtls_cid);
+				ret = -ENOTSUP;
+				goto cli_exit;
```

Comment:
Why do we fail instead of just print out the status?

It is up to server configuration to accept the use of CID or not.

Reply:
oh... I did not know that there is this weird distinction about socket commands and UDP client commands...